qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -266,63 +251,198 @@ def run(self, dag):
         output_layout = Layout({dag.qubits[k]: v for (k, v) in layout_mapping})
         self.property_set["final_layout"] = output_layout
         if not self.fake_run:
-            for node_id in gate_order:
-                node = dag._multi_graph[node_id]
-                process_swaps(
-                    swap_map,
-                    node,
-                    mapped_dag,
-                    original_layout,
-                    canonical_register,
-                    self.fake_run,
-                    self._qubit_indices,
+            mapped_dag = dag.copy_empty_like()
+            _apply_sabre_result(
+                mapped_dag,
+                dag,
+                self._qubit_indices,
+                canonical_register,
+                original_layout,
+                sabre_result,
+                circuit_to_dag_dict,
+            )
+            return mapped_dag
+        return dag
+
+
+def _build_sabre_dag(dag, num_qubits, num_clbits, qubit_indices, clbit_indices):
+    from qiskit.converters import circuit_to_dag
+
+    # Maps id(block): circuit_to_dag(block) for all descendant blocks
+    circuit_to_dag_dict = {}
+
+    def recurse(block, block_qubit_indices, block_clbit_indices):
+        block_dag = circuit_to_dag(block)
+        circuit_to_dag_dict[id(block)] = block_dag
+        return process_dag(block_dag, block_qubit_indices, block_clbit_indices)
+
+    def process_dag(block_dag, block_qubit_indices, block_clbit_indices):
+        dag_list = []
+        node_blocks = {}
+        for node in block_dag.topological_op_nodes():
+            cargs = {block_clbit_indices[x] for x in node.cargs}
+            if node.op.condition is not None:
+                for clbit in block_dag._bits_in_condition(node.op.condition):
+                    cargs.add(block_clbit_indices[clbit])
```

Comment:
I'll need to modify this slightly to support #10232, but there's nothing better to do in this PR.